### DISCUSSION TITLE
Renderer crash because of Memory Allocation

### DISCUSSION THREAD

**Discussion by JanLochi** [Igalia/meta-webkit#458] [Q&A] [answered]
HI All.

I have an i.MX6 with Yocto Dunfell and the current main branch of meta-webkit in use.

cog is in started with ` cog -P drm --platform-params=renderer=gles` and uses the etnaviv driver.

While starting the cog browser, I sometimes get the error:

```
DRM_IOCTL_MODE_CREATE_DUMB failed: Cannot allocate memory
Crash!: The renderer process crashed. Reloading the page may fix intermittent failures.
```

Are there some Kernel configs to improve the situation or similar?

Thx!

**Comment by philn** [comment]
CCing @psaavedra who's worked on etnaviv/wpe at some point in the recent past ;)

**Comment by JanLochi** [comment]
Addendum: My target device has 256MB RAM and one CPU core. Its sole job is to host the cog browser.

**Comment by JanLochi** [comment, marked as answer]
Sooo, after investing some hours on that issue. I discovered, that my BSP has the Kernel Config: `CONFIG_CMA_SIZE_MBYTES=320`.
With a SOM with only 256MB of RAM, this consequently leads to a CMA pool of nothing.

Most likely, with a sane setting of `CONFIG_CMA_SIZE_MBYTES`, the cog browser should work as expected.

**Reply by philn** [reply]
Please let us know how it goes after fixing the CMA size!

**Reply by JanLochi** [reply]
Looks like it is working now. :+1: 

The CMA is now set to 64MB.

Any suggestions to have cog running on that resource constrained device? Should I use these params?
```
--web-mem-limit
--net-mem-limit
```

I have some other issues now, I will open another discussion to not clutter this one.
